From: Tzafrir Cohen <tzafrir@debian.org>
Subject: remove early homedir generation in Debian

Various rpm-related tools generate the home directory of the mockbuild
user in the chroot (/buildroot) too soon by leaving a ~/.rpmdb directory
there. This prevents useradd from giving it proper permissions and the
install fails.

Remove this directory.

--- a/py/mockbuild/buildroot.py
+++ b/yp/mockbuild/buildroot.py
@@ -221,6 +221,10 @@ class Buildroot(object):
 
         self.doChroot(['/usr/sbin/groupadd', '-g', dets['gid'], dets['group']],
                       shell=False, nosync=True)
+        homedir = self.make_chroot_path(self.homedir)
+        if os.path.exists(homedir):
+            util.rmtree(os.path.join(homedir, '.rpmdb')) # If it exists
+            os.rmdir(homedir)
         self.doChroot(shlex.split(self.config['useradd'] % dets), shell=False, nosync=True)
         self._enable_chrootuser_account()
 
